src/libraries/Microsoft.PowerFx.Connectors/Public/CdpRecordType.cs

```diff
@@ -94,6 +98,13 @@ public override int GetHashCode()
 
         public override string TableSymbolName => ConnectorType.Name;
 
-        public override IEnumerable<string> FieldNames => _type.GetRootFieldNames().Select(name => name.Value);
+        public override IEnumerable<string> FieldNames
+        {
+            get
+            {
+                _fieldNames ??= ConnectorType.Fields.Select(field => field.Name).ToList();
```

Comment:
>_fieldNames [](http://example.com/codeflow?start=16&length=11)

does caching help us here?

Reply:
Getting the list of field names is a frequently used operation in this code and this optimization seems useful even if I can't provide performance numbers to measure this

Reply:
this is a cache on a cache.